tests/TestCase/Console/CommandRunnerTest.php

```diff
@@ -526,6 +528,36 @@ public function testRunLoadsRoutes(): void
         $this->assertGreaterThan(2, count(Router::getRouteCollection()->routes()));
     }
 
+    /**
+     * Test the Console.afterRun event
+     */
+    public function testAfterRunEvent(): void
+    {
+        $app = $this->getMockBuilder(BaseApplication::class)
+            ->onlyMethods(['middleware', 'bootstrap', 'routes'])
+            ->setConstructorArgs([$this->config])
+            ->getMock();
+
+        $output = new ConsoleOutput();
+        $runner = new CommandRunner($app, 'cake');
+        $runner->getEventManager()->on('Console.afterRun', function ($event, $command, $result) {
+            if ($command instanceof VersionCommand && $result === 0) {
+                return [
+                    [
+                        'command' => CacheClearCommand::class,
+                        'arguments' => ['_cake_model_'],
+                    ],
+                ];
```

Comment:
Should event listeners use `CommandRunner` directly if they need to invoke additional commands? Having an array 'meta' format for invoking other commands is not a pattern we share with other callbacks in the framework.

Reply:
I wasn't sure about that as well but I thought this rather "simple" array structure is self-explanatory.
The main functionality I was "copying" was the way how one can execute commands inside other commands.
https://book.cakephp.org/4/en/console-commands/commands.html#calling-other-commands

But I also didn't want to just restrict the user to just 1 command, therefore the array.